test/blockchain/vbk_blockchain_util_test.cpp

```diff
@@ -0,0 +1,191 @@
+            VbkChainParamsMain().getRetargetPeriod() + 1
+        },
+        {
+            ArithUint256::fromHex("09FFF600000000000000000000000000000000000000").toBits(),
+            0,
+            0,
+            ArithUint256::fromHex("00FFFF00000000000000000000000000000000000000").toBits(),
+            VbkChainParamsMain().getRetargetPeriod() + 1
+        },
+        {
+            ArithUint256::fromHex("02FFFD00000000000000000000000000000000000000").toBits(),
+            (int32_t) VbkChainParamsMain().getTargetBlockTime() / 3,
+            0,
+            ArithUint256::fromHex("00FFFF00000000000000000000000000000000000000").toBits(),
+            VbkChainParamsMain().getRetargetPeriod() + 1
+        },
+        {
+            ArithUint256::fromHex("09FFF600000000000000000000000000000000000000").toBits(),
+            -10,
+            0,
+            ArithUint256::fromHex("00FFFF00000000000000000000000000000000000000").toBits(),
+            VbkChainParamsMain().getRetargetPeriod() + 1
+        },
+        {
+            ArithUint256::fromHex("09C23793D191").toBits(),
+            (int32_t) VbkChainParamsMain().getTargetBlockTime() + 10,
+            10,
+            ArithUint256::fromHex("09184E72A000").toBits(),
+            VbkChainParamsMain().getRetargetPeriod() + 1
+        },
+        {
+            ArithUint256::fromHex("08840877ADF0").toBits(),
+            (int32_t) VbkChainParamsMain().getTargetBlockTime() - 10,
+            -10,
+            ArithUint256::fromHex("09184E72A000").toBits(),
+            VbkChainParamsMain().getRetargetPeriod() + 1
+        }
+        // clang-format on
+};
+
+TEST_P(GetNextWorkRequiredTest, getNextWorkRequired_test) {
+  auto value = GetParam();
+
+  std::vector<BlockIndex<VbkBlock>> chain = getChain(value.deltaTime,
+                                                     value.deltaTime_change,
+                                                     value.chain_difficulty,
+                                                     value.chainlength);
+
+  uint32_t result = getNextWorkRequired(
+      chain[chain.size() - 2], chain[chain.size() - 1].header, *chainparams);
+
+  EXPECT_EQ(value.expected_difficulty, result);
+}
+
+INSTANTIATE_TEST_SUITE_P(GetNextWorkRequiredRegression,
+                         GetNextWorkRequiredTest,
+                         testing::ValuesIn(getNextWorkRequired_test_cases));
+struct SingleTest : public ::testing::Test, public VbkBlockchainUtilTest {};
+
+TEST_F(SingleTest, single_test) {
+  uint32_t chainlength = chainparams->getRetargetPeriod() + 1;
+
+  int32_t deltaTime = chainparams->getTargetBlockTime();
+
+  BlockIndex<VbkBlock> blockIndex;
+  blockIndex.header.height = 1;
+  blockIndex.header.timestamp = 10000;
+  blockIndex.header.difficulty = ArithUint256::fromHex("09184E72A000").toBits();
+  blockIndex.height = blockIndex.header.height;
+  blockIndex.pprev = nullptr;
+
+  std::vector<BlockIndex<VbkBlock>> chain(chainlength);
```

Comment:
I think you could use getChain() here.

Reply:
Here is used another algorithm of the chain building that is hard to unify to all purposes of the previous tests